### ISSUE TITLE
Variable references in DataFrame.query not recognized as such

### ISSUE THREAD

**Issue by ymyke** [PyCQA/pylint#3903]
### Problem description

Variable names in Pandas queries are not recognized as variable references by Pylint and produce `unused-variable` (unless used outside the Pandas query obviously).

E.g., in: `query = "somecolumn == @something"`

### Possible solution

Make Pylint recognize @-prefixed parts in strings and consider those as variable uses. This approach might lead to false positives. A more diligent approach might be to only do so when pandas is imported and/or the string in question is used in a `DataFrame.query()` call.

### Notes

* I'm surprised I didn't find anything about this in the issues, but I did have a look.
* Similar issue with `eval`?

**Comment by hippo91** [comment]
@ymyke i'm afraid this would require much more work than we are able to produce.
Can you give a precise example of the problem? (i.e a piece of code that we will be able to lint)?

**Comment by ymyke** [comment]
@hippo91 Here's a toy example that illustrates the issue:

```python
import pandas as pd


def print_row(df, row):
    print(df.query("row == @row"))


print_row(pd.DataFrame({"row": [1, 2], "X": [3, 4]}), 2)
```

**Comment by ymyke** [comment]
And here's a (silly) toy example that illustrates the issue with `eval`:

```python
def print_square(number):
    print(eval("number ** 2"))

print_square(2)
```

**Comment by hippo91** [comment]
@ymyke thanks for those clear examples!
